Cubical/Data/FinSet.agda

```diff
@@ -0,0 +1,110 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Data.FinSet where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Equiv
+open import Cubical.HITs.PropositionalTruncation
+open import Cubical.Data.Unit
+open import Cubical.Data.Nat
+open import Cubical.Data.Fin
+open import Cubical.Data.Sigma
+
+private
+  variable
+    ℓ : Level
+    A : Type ℓ
+
+isFinSet : Type ℓ → Type ℓ
+isFinSet A = ∃[ n ∈ ℕ ] A ≃ Fin n
+
+isProp-isFinSet : isProp (isFinSet A)
+isProp-isFinSet = propTruncIsProp
+
+FinSet : Type (ℓ-suc ℓ)
+FinSet = TypeWithStr _ isFinSet
+
+isFinSetFin : ∀ {n} → isFinSet (Fin n)
+isFinSetFin = ∣ _ , pathToEquiv refl ∣
+
+isFinSetUnit : isFinSet Unit
+isFinSetUnit = ∣ 1 , Unit≃Fin1 ∣
+  where
+    Unit≃Fin1 : Unit ≃ Fin 1
+    Unit≃Fin1 =
+      isoToEquiv
+        (iso
+          (const fzero)
+          (const tt)
+          (isContrFin1 .snd)
+          (isContrUnit .snd)
+        )
+
+isFinSet′ : Type ℓ → Type ℓ
+isFinSet′ A = Σ[ n ∈ ℕ ] ∥ A ≃ Fin n ∥
+
+FinSet′ : Type (ℓ-suc ℓ)
+FinSet′ = TypeWithStr _ isFinSet′
+
+isProp-isFinSet′ : isProp (isFinSet′ A)
+isProp-isFinSet′ {A = A} (n , equivn) (m , equivm) =
+  Σ≡Prop (λ _ → propTruncIsProp) n≡m
+  where
+    Fin-n≃Fin-m : ∥ Fin n ≃ Fin m ∥
+    Fin-n≃Fin-m = rec
+      propTruncIsProp
+      (rec
+        (isPropΠ λ _ → propTruncIsProp)
+        (λ hm hn → ∣ Fin n ≃⟨ invEquiv hn ⟩ A ≃⟨ hm ⟩ Fin m ■ ∣)
+        equivm
+      )
+      equivn
+
+    Fin-n≡Fin-m : ∥ Fin n ≡ Fin m ∥
+    Fin-n≡Fin-m = rec propTruncIsProp (∣_∣ ∘ ua) Fin-n≃Fin-m
+
+    ∥n≡m∥ : ∥ n ≡ m ∥
+    ∥n≡m∥ = rec propTruncIsProp (∣_∣ ∘ Fin-inj n m) Fin-n≡Fin-m
+
+    n≡m : n ≡ m
+    n≡m = rec (isSetℕ n m) (λ p → p) ∥n≡m∥
+
+isFinSet≡isFinSet′ : isFinSet A ≡ isFinSet′ A
+isFinSet≡isFinSet′ {A = A} = ua (isoToEquiv (iso to from to-from from-to))
```

Comment:
You can use https://github.com/agda/cubical/blob/master/Cubical/Foundations/Univalence.agda#L42 here